File: fullstack-challenge/backend-ts/src/domain/transaction.ts
export interface Transaction {
  id: string;
  deviceId: string;
  counter: number;
  timestamp: Date;
  data: string;
  signature: string;        // Base64 encoded
  signedData: string;       // The full string that was signed
  previousSignature: string; // Base64 encoded or base64(device.id) for first transaction
}

export interface CreateTransactionInput {
  deviceId: string;
  data: string;
}

// TODO: Add validation and business logic for Transaction

File: fullstack-challenge/backend-go/domain/transaction.go
package domain

import "time"

// Transaction represents a signed piece of data.
type Transaction struct {
	ID                string
	DeviceID          string
	Counter           int
	Timestamp         time.Time
	Data              string
	Signature         string // Base64 encoded
	SignedData        string // The full string that was signed
	PreviousSignature string // Base64 encoded or base64(device.id) for first transaction
}

// TODO: Add validation methods and business logic for Transaction
